src/style/style.js

```diff
@@ -1120,6 +1120,7 @@ class Style extends Evented {
         for (const sourceCache of sourceCaches) {
             sourceCache.setFeatureState(sourceLayer, target.id, state);
         }
+        this.fire(new Event('setfeaturestate'));
```

Comment:
I don;t think we need an event for this one, you could directly do `this.map.painter.clearTerrainRenderCache()` (this method doesnt exist on `Painter` but you could add it.